tests/cross/src/test/scala/tests/pc/CompletionPatternSuite.scala

```diff
@@ -60,7 +60,25 @@ class CompletionPatternSuite extends BaseCompletionSuite {
       |    case ma@@
       |  }
       |}""".stripMargin,
-    ""
+    "",
+    compat = Map(
+      scala3PresentationCompilerVersion ->
+        """|main scala
```

Comment:
So this would show before if we wrote for example `mai` ? I think we shouldn't show anything in this case, but might be a different bug, not to fix here.

Reply:
yes

Reply:
would be good to fix in a separate issue, though it seems like something that we dealt before with? @jkciesluk ?

Reply:
It's shown here because `Completions.advancedCompletions` for patterns in `match-case` are not marked as `exclusive` (if they were, we wouldn’t suggest eg. objects with unapply methods).

Reply:
Could we make sure only things with unapply show up?

Reply:
Yeah, some time ago I was trying to implement completions for them that would match the result type of the `unapply` method, but didn’t manage to finish it. Adding just completion for object where `unapply` is defined (with maybe not matching parameter type) should be a lot easier.

Reply:
That would already be better I think, but anyway, not related

Reply:
Maybe we can leave this `compat` here now and remove after adding this feature?

Reply:
https://github.com/scalameta/metals/issues/5972